src/geometry/ArborX_HyperPoint.hpp

```diff
@@ -22,6 +22,8 @@ namespace ArborX::ExperimentalHyperGeometry
 template <int DIM, class FloatingPoint = float>
 struct Point
 {
+  static_assert(std::is_floating_point_v<FloatingPoint>);
```

Comment:
I don't think we want this. It should still be possible for a user to build `Point<D, int>` if desired.

Reply:
Then, we shouldn't call the template argument `FloatingPoint`.

Reply:
I agree in principle. Not changing as part of this PR, though.